### PR TITLE
[SPARK-31086][SQL] Add Back the Deprecated SQLContext methods

### PR DESCRIPTION
**Pull request by gatorsmile**
### What changes were proposed in this pull request?

Based on the discussion in the mailing list [[Proposal] Modification to Spark's Semantic Versioning Policy](http://apache-spark-developers-list.1001551.n3.nabble.com/Proposal-Modification-to-Spark-s-Semantic-Versioning-Policy-td28938.html) , this PR is to add back the following APIs whose maintenance cost are relatively small.

- SQLContext.applySchema
- SQLContext.parquetFile
- SQLContext.jsonFile
- SQLContext.jsonRDD
- SQLContext.load
- SQLContext.jdbc

### Why are the changes needed?
Avoid breaking the APIs that are commonly used.

### Does this PR introduce any user-facing change?
Adding back the APIs that were removed in 3.0 branch does not introduce the user-facing changes, because Spark 3.0 has not been released. 

### How was this patch tested?
The existing tests.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala [hunk 1]
```diff
@@ -611,6 +611,289 @@ class SQLContext private[sql](val sparkSession: SparkSession)
...
+  ////////////////////////////////////////////////////////////////////////////
+  ////////////////////////////////////////////////////////////////////////////
+  // Deprecated methods
+  ////////////////////////////////////////////////////////////////////////////
+  ////////////////////////////////////////////////////////////////////////////
+
+  /**
+   * @deprecated As of 1.3.0, replaced by `createDataFrame()`.
+   */
+  @deprecated("Use createDataFrame instead.", "1.3.0")
```
Comment (dongjoon-hyun) [inline sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala]: Hi, @marmbrus and @srowen .
Do you really need to keep `1.3.0` API?

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Hi, All (including reviewers).
Although this is marked as `Draft` clearly, I want to make it sure that we need to hold on this until we have a vote.

File: sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala (hunk 1 above)
Comment (srowen) [reply sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala]: I agree; I also don't feel strongly about it as there isn't much overhead to keeping it, beyond API noise. 
@gatorsmile may have some additional info or data about this.

**Comment by srowen** [comment]
Yes, is it simpler to just list the things being proposed to change and why (i.e. any data or arguments to support it) first? It does seem like a whole lot is being added back, which is at odds with what I understand to be the conclusion of the last thread on this.

**Comment by dongjoon-hyun** [comment]
Also, cc @marmbrus .
And, cc @rxin since he is the release manager for 3.0.0.

File: sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala (hunk 1 above)
Comment (marmbrus) [reply sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala]: The rubric says nothing about how old the deprecation warning is on purpose. It says we should think about usage.

@gatorsmile do you have any reason to believe these are commonly used functions?

Comment (gatorsmile) [reply sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala]: At least these Spark books are still using this API in their examples, although the first two claims they are updated using Spark 2.x

- https://www.amazon.com/Mastering-Apache-Spark-2-x-DeepLearning4j/dp/1786462745/ref=sr_1_4?keywords=mastering+Apache+Spark&qid=1584678721&s=books&sr=1-4
- https://www.manning.com/books/spark-in-action
- http://shop.oreilly.com/product/0636920028512.do [first edition]

**Comment by gatorsmile** [comment]
CC @dongjoon-hyun @marmbrus @rxin @srowen This PR is ready for review.

**Review by srowen — COMMENTED** [review]
I don't feel strongly about this either way. I think these APIs are going to be here forever if kept here now, and I don't see evidence they are used, so, why were they even deprecated and replaced? Deprecation loses meaning. But, they impose almost no maintenance burden as aliases, so "who cares?" can cut either way.

**Review by dongjoon-hyun — APPROVED** [review]

**Review by cloud-fan — APPROVED** [review]
LGTM as these are easy to maintain.

**Comment by gatorsmile** [comment]
Thanks! Merged to master/3.0

**Pull request closed without merging**